### ISSUE TITLE
Compile in docker build server causes overlayfs error 

### ISSUE THREAD

**Issue by Kiwi-MethodPark** [mozilla/sccache#584]
I used to compile in a build server created with the following docker file:
```
FROM rust

RUN wget http://http.us.debian.org/debian/pool/main/b/bubblewrap/bubblewrap_0.3.3-2_amd64.deb
RUN dpkg -i bubblewrap_0.3.3-2_amd64.deb

RUN git clone https://github.com/mozilla/sccache.git
RUN cd sccache && \ 
    cargo build --features=dist-server --bin=sccache-dist --package=sccache

ENTRYPOINT [ "./sccache/target/debug/sccache-dist" ]
```
Unfortunately it results in a compilation error due to file system overlay, is there a workaround/solution?

sccache::dist::http::server: Res 2 error: run build failed, caused by: Compilation execution failed, caused by: Failed to mount overlay FS, caused by: overlayfs "/tmp/build/tool-chains/114b65dde7522750cbe7a4f187808058436e89652fd37241ed589afc0423a6f5bf414d2060d48ba6fc7679a0ea2d7cc7db750c41aee6cc946a894d43dcf4609d",upperdir="/tmp/build/builds/114b65dde7522750cbe7a4f187808058436e89652fd37241ed589afc0423a6f5bf414d2060d48ba6fc7679a0ea2d7cc7db750c41aee6cc946a894d43dcf4609d-1/upper",workdir="/tmp/build/builds/114b65dde7522750cbe7a4f187808058436e89652fd37241ed589afc0423a6f5bf414d2060d48ba6fc7679a0ea2d7cc7db750c41aee6cc946a894d43dcf4609d-1/work" -> "/tmp/build/builds/114b65dde7522750cbe7a4f187808058436e89652fd37241ed589afc0423a6f5bf414d2060d48ba6fc7679a0ea2d7cc7db750c41aee6cc946a894d43dcf4609d-1/target": Invalid argument (os error 22) ("/tmp/build/tool-chains/114b65dde7522750cbe7a4f187808058436e89652fd37241ed589afc0423a6f5bf414d2060d48ba6fc7679a0ea2d7cc7db750c41aee6cc946a894d43dcf4609d": exists, upperdir: exists, workdir: exists, same-fs, target: exists, superuser)

**Comment by vklindukh** [comment]
run docker with `-v /tmp:/tmp`

**Comment by AtomicOperation** [comment]
For future reference, a better way to do this is to use the **tmpfs** option to docker run or in docker_compose.yml. I think that bind-mounting /tmp will only work on Linux host systems where /tmp is itself tmpfs, whereas creating a tmpfs mount inside the container works on other platforms as well.

Also, the quick summary of the reason this is necessary (as I understand it) is that overlay mounts, which sccache uses to containerize the builds it does, can't be made writable under docker on 'normal' filesystems because the docker container itself is set up with a writable overlayfs and there can be only one. For some reason this limitation doesn't apply if the filesystem that's being overlaid is tmpfs.

**Comment by Kiwi-MethodPark** [comment]
Issue is solved when working with podman version 3.4.4 and the current sccache code. 
Don't know who "fixed" the issue but it will work know, thank's
